pkg/utils/vault/secret_file.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package vault
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	// inMemory is the name of the file to use to not write it to disk and only keep it in memory
+	inMemory = "in-memory"
+
+	// buildLicensePubKeyPrefixEnvVar allows to prefix the field to retrieve a specific license public key secret
+	buildLicensePubKeyPrefixEnvVar = "BUILD_LICENSE_PUBKEY"
+
+	retryTimeout  = 10 * time.Second
+	retryInterval = 1 * time.Second
+)
+
+// SecretFile maps a Vault Secret into a file that is optionally written to disk.
+type SecretFile struct {
+	// Name is the name of the file to optionaly write the secret to disk. If the value is 'in-memory' the file is not written to disk.
+	Name string
+	// Path is the Vault path to read the secret
+	Path string
+	// FormatJSON indicates if the secret needs to be printed in JSON format. It can be only true if FieldResolver is not set.
+	FormatJSON bool
+	// FieldResolver is a function to get only a given field of the secret. It is optional and can be nil.
+	// It can be only set if FormatJSON is false.
+	FieldResolver func() string
+	// Base64Encoded indicates if the secret needs to be decoded. It is only usable with FieldResolver is set.
+	Base64Encoded bool
+}
+
+// ReadFile reads the file if it exists or else reads the Secret from Vault and
+// writes it to the file on disk.
+func ReadFile(c Client, f SecretFile) ([]byte, error) {
+	if _, err := os.Stat(f.Name); err == nil {
+		return os.ReadFile(f.Name)
+	}
+
+	bytes, err := f.readFromVault(c)
+	if err != nil {
+		return nil, err
+	}
+
+	if f.Name != inMemory {
+		err = os.WriteFile(f.Name, bytes, 0600)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return bytes, nil
+}
+
+func (f SecretFile) readFromVault(c Client) ([]byte, error) {
+	log.Printf("Read %s from vault", f.Path)
```

Comment:
Do we want to keep this?

Reply:
I wasn't sure, it's useful but it prints a log line which may seem to come from nowhere. Let's delete it.